src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -85,22 +85,52 @@ public IEnumerable<CompletionItem> GetFilteredCompletions(Compilation compilatio
 
         public IEnumerable<CompletionItem> GetFilteredParamsCompletions(ParamsSemanticModel paramsSemanticModel, ParamsCompletionContext paramsCompletionContext)
         {
-            return GetParamAssignmentCompletions(paramsSemanticModel, paramsCompletionContext);
+            return GetParamAssignmentCompletions(paramsSemanticModel, paramsCompletionContext)
+                  .Concat(GetUsingDeclarationPathCompletions(paramsSemanticModel, paramsCompletionContext));
         }
 
         private IEnumerable<CompletionItem> GetParamAssignmentCompletions(ParamsSemanticModel paramsSemanticModel, ParamsCompletionContext paramsCompletionContext)
         {   
             if(paramsCompletionContext.Kind.HasFlag(ParamsCompletionContextKind.ParamAssignment) && paramsSemanticModel.BicepCompilation is {} bicepCompilation)
             {
-                //TODO: ignore params already declared
                 var bicepSemanticModel = bicepCompilation.GetEntrypointSemanticModel();
                 var bicepFileParamDeclarations = bicepSemanticModel.Root.ParameterDeclarations;
 
                 foreach(var declaration in bicepFileParamDeclarations)
                 {
-                    yield return CreateSymbolCompletion(declaration, paramsCompletionContext.ReplacementRange);
+                    if(!IsParamAssigned(declaration))
+                    {
+                        yield return CreateSymbolCompletion(declaration, paramsCompletionContext.ReplacementRange);
+                    }
                 }
             }
+
+            bool IsParamAssigned(ParameterSymbol declaration) => paramsSemanticModel.ParamBinder.ParamFileSymbol.ParameterAssignmentSymbols.Any(paramDeclaration => paramDeclaration.Name == declaration.Name);
+        }
+
+        private IEnumerable<CompletionItem> GetUsingDeclarationPathCompletions(ParamsSemanticModel paramsSemanticModel, ParamsCompletionContext paramsCompletionContext)
+        {
+            if (!paramsCompletionContext.Kind.HasFlag(ParamsCompletionContextKind.UsingDeclaration))
+            {
+                return Enumerable.Empty<CompletionItem>();
+            }
+
+            var entered = "";
```

Comment:
You'll probably need to copy the implementation here:
https://github.com/Azure/bicep/blob/9a45fd99264df606f4fb36ef660ad1372b3f7729/src/Bicep.LangServer/Completions/BicepCompletionProvider.cs#L458-L464

If you want the following scenario (searching while typing) to work:

https://user-images.githubusercontent.com/38542602/180342381-ef85af88-085a-4595-930c-468dac2b1cd5.mov